slack/web/classes/interactions.py

```diff
@@ -0,0 +1,123 @@
+import json
+from typing import List
+
+from .objects import IDNamePair, JsonObject
+
+
+class InteractiveEvent(JsonObject):
+    response_url: str
+    user: IDNamePair
+    team: IDNamePair
+    channel: IDNamePair
+
+    raw_event: dict
+
+    def __init__(self, event: dict):
+        self.raw_event = event
+        self.response_url = event["response_url"]
+
+
+class MessageInteractiveEvent(InteractiveEvent):
+    event_type: str
+    message_ts: str
+    trigger_id: str
+    action_id: str
+    block_id: str
+    message: dict
+
+    def __init__(self, event: dict):
+        """
+        Convenience class to parse an interactive message payload from the events API
+        :param event: the raw event dictionary
+        """
+        super().__init__(event)
+        self.user = IDNamePair(event["user"]["id"], event["user"]["username"])
+        self.team: IDNamePair = IDNamePair(event["team"]["id"], event["team"]["domain"])
+        self.channel: IDNamePair = IDNamePair(
+            event["channel"]["id"], event["channel"]["name"]
+        )
+        self.event_type = event["type"]
+        self.message_ts = event["message"]["ts"]
+        self.trigger_id = event["trigger_id"]
+        action = event["actions"][0]
+        self.action_id = action["action_id"]
+        self.block_id = action["block_id"]
+        if action.get("selected_option"):
+            self.value = action["selected_option"]["value"]
+        else:
+            self.value = action["value"]
+        self.message = event["message"]
+
+
+class DialogInteractiveEvent(InteractiveEvent):
+    event_type: str
+    submission: dict
+    state: dict
+
+    def __init__(self, event: dict):
+        """
+        Convenience class to parse a dialog interaction payload from the events API
```

Comment:
These convenience classes are definitely handy. Once https://github.com/slackapi/python-slackclient/issues/403 is picked up I'd imagine we could bake these in directly.